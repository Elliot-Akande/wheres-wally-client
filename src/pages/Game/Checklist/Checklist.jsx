import ChecklistItem from "../ChecklistItem/ChecklistItem";

const Checklist = ({ characters, correctAnswers }) => {
  const isFound = (character) => {
    return Object.prototype.hasOwnProperty.call(correctAnswers, character.name);
  };

  return (
    <>
      <h2>Checklist:</h2>
      <ul>
        {characters.map((character) => (
          <ChecklistItem
            key={character.name}
            character={character.name}
            found={isFound(character)}
          />
        ))}
      </ul>
    </>
  );
};

export default Checklist;
