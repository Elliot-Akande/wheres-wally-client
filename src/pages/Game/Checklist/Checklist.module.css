.list {
  padding: 0;

  display: flex;
  gap: 10px;

  list-style: none;
}
