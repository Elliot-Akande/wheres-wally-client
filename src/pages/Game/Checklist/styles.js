import { styled } from "styled-components";

export const StyledChecklist = styled.ul`
  margin: 0;
  padding: 0;

  display: flex;
  flex-wrap: wrap;
  gap: 8px;

  list-style: none;
`;
