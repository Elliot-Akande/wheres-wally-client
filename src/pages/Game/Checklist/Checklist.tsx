import PropTypes from "prop-types";
import ChecklistItem from "../ChecklistItem/ChecklistItem";
import { StyledChecklist } from "./styles";

const Checklist = ({ characters, correctAnswers }) => {
  const isFound = (character) => {
    return correctAnswers.find((answer) => answer.character === character);
  };

  return (
    <StyledChecklist>
      {characters.map((character) => (
        <ChecklistItem
          key={character.name}
          character={character}
          checked={!!isFound(character.name)}
        />
      ))}
    </StyledChecklist>
  );
};

Checklist.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      imageUrl: PropTypes.string.isRequired,
    })
  ),
  correctAnswers: PropTypes.arrayOf(
    PropTypes.shape({
      character: PropTypes.string,
    })
  ).isRequired,
};

export default Checklist;
