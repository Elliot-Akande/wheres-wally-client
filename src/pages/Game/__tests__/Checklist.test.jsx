import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import Checklist from "../Checklist/Checklist.jsx";

vi.mock("../ChecklistItem/ChecklistItem.jsx", () => ({
  default: ({ character, checked }) => (
    <li>
      {character.name} is {checked ? "Found" : "Not Found"}
    </li>
  ),
}));

const characters = [
  { name: "Harry", id: 1, imageUrl: "harry.jpg" },
  { name: "Hermione", id: 2, imageUrl: "hermione.jpg" },
];
const correctAnswers = [{ character: "Hermione" }];

describe("Checklist", () => {
  it("renders a list of characters with correct checked status", () => {
    const { container } = render(
      <Checklist characters={characters} correctAnswers={correctAnswers} />
    );

    expect(container).toMatchSnapshot();
  });
});
