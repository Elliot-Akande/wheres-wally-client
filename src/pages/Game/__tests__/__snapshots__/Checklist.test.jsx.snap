// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checklist > renders a list of characters with correct checked status 1`] = `
<div>
  <ul
    class="sc-beySPh iukuLq"
  >
    <li>
      Harry
       is 
      Not Found
    </li>
    <li>
      Hermione
       is 
      Found
    </li>
  </ul>
</div>
`;
